### PR TITLE
#184637250 Setup Authentication with Google

### PR DESCRIPTION
**Pull request by dean-daryl**
**What does this PR do?**

-Setup Authentication with Google

**Description of Task to be completed?**

- Added Authentication using Google
- A token is generated using JWT when a user successfully logs in

**How should this be manually tested?**

- clone the project and use the terminal to start the development server by "npm run dev"
- run on localhost:5000 

**Any background context you want to provide?**
none

**What are the relevant pivotal tracker/Trello stories?**

- Tracker:184637250

**Screenshots (if appropriate)**
none

**Questions:**
none

### CODE REVIEW

File: src/config/passport.js [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import { Strategy as GoogleStrategy } from 'passport-google-oauth2';
+
+dotenv.config();
+
+passport.use(
+  new GoogleStrategy(
+    {
+      clientID: process.env.GOOGLE_CLIENT_ID,
+      clientSecret: process.env.GOOGLE_CLIENT_SECRET,
+      callbackURL: `http://localhost:${process.env.PORT}/google/callback`,
```
Comment (fab-ryan) [inline src/config/passport.js]: make this variable from process environmental

File: src/config/passport.js [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+      passReqToCallback: true,
+    },
+    async (request, accessToken, refreshToken, profile, done) => {
+      // console.log(profile.email)
+      done(null, profile);
+
+      //  User.findOrCreate({ googleId: profile.id }, function (err, user) {
+      //    return done(err, user);
+      //  });
```
Comment (fab-ryan) [inline src/config/passport.js]: remove those commented lines

File: src/server.js [hunk 3]
```diff
@@ -1,13 +1,70 @@
...
+);
+
+app.get('/google/callback', googleCallback, async (req, res) => {
+  try {
+    const { given_name, email } = req.user;
+    const googleUser = await User.findOne({ where: { email } });
+    let userObject = {};
+    if (googleUser) {
+      const TOKEN = jwt.sign(
+        { name: given_name, email },
```
Comment (fab-ryan) [inline src/server.js]: create helper function that can generate token because you team they might need in  future

File: src/server.js [hunk 4]
```diff
@@ -1,13 +1,70 @@
...
+      );
+      const user = await User.create({
+        name: given_name,
+        email,
+        password: BcryptUtil.hash('password'),
+      });
+      user.save();
+      userObject = { name: given_name, token: TOKEN };
+    }
+    res.status(200).json({ userObject });
```
Comment (fab-ryan) [inline src/server.js]: try to return meaning response

**Comment by istafrica-dennw** [comment]
Please make a WIP, as a feature is ready when tested! and you have not yet written any test,

File: src/config/passport.js (hunk 1 above)
Comment (fab-ryan) [reply src/config/passport.js]: you haven't changed this

**Review by kirengaboris — APPROVED** [review]

**Review by masterchief-00 — APPROVED** [review]

**Review by istafrica-dennw — APPROVED** [review]

**Review by fab-ryan — APPROVED** [review]

**Pull request merged**